### ISSUE TITLE
swift can't override db_log and db_save

### ISSUE THREAD

**Issue by changanli** [CocoaLumberjack/CocoaLumberjack#1005]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [ ] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / osx / tvos / watchos
 Platform Version        | e.g. 8.0
 CocoaLumberjack Version | e.g. 2.3.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps
class Logger :DDAbstractDatabaseLogger
swift can't override db_log and db_save
Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by changanli** [comment]
override func db_log(_ logMessage: DDLogMessage!) -> Bool {
        if self.logFormatter == nil {
            //没有指定 formatter
            return false
        }

return false
}

Crashed !!!
